reindexer/reindex_worker/src/main/scala/uk/ac/wellcome/platform/reindex/reindex_worker/dynamo/BatchItemGetter.scala

```diff
@@ -0,0 +1,28 @@
+package uk.ac.wellcome.platform.reindex.reindex_worker.dynamo
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB
+import com.amazonaws.services.dynamodbv2.document._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+class BatchItemGetter(dynamoDBClient: AmazonDynamoDB)(
+  implicit ec: ExecutionContext) {
+
+  final val PARTITION_KEY: String = "id"
+  val dynamoDB = new DynamoDB(dynamoDBClient)
+
+  def get(recordIds: List[String])(tableName: String): Future[List[String]] =
```

Comment:
Would be good to get a brief comment on this class explaining what it does, and how it’s used.

Would be good to mention:

* What if I request a really big list of IDs?
* What if I request an ID that doesn't exist?

Naming nits:

* The class is called `BatchItemGetter` but you pass it a list of `recordIds`. Would it be better to pass a list of `itemIds`?
* Everywhere it’s used, it’s called `SpecificItemGetter`. Would that be neater?
* Where does the term `PARTITION_KEY` come from? In DynamoDB we have hash keys; I'm not familiar with that term.

Reply:
Good points, will add documentation comment.

- yep good point re `recordIDs`
- went for `BatchItemGetter` to reflect the name of the DynamoDB operation (as with `ScanSpecScanner`) given that it directly wraps the client. Will probably leave this for now if that's OK with you.
- yeah, I've had this issue before, I have a feeling AWS changed their naming convention at some point: now they almost exclusively refer to partition keys (eg [here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.CoreComponents.html)) although some APIs call them hash keys 🤷‍♂